### PR TITLE
[URLSession]Fixing a warning and some improvements over PR 968

### PR DESCRIPTION
**Pull request by pushkarnk**

### CODE REVIEW

**Comment by pushkarnk** [comment]
@nethraravindran Can you review this please?

File: Foundation/NSURLSession/NSURLSessionTask.swift [hunk 1]
```diff
@@ -543,13 +543,14 @@ extension _ProtocolClient : URLProtocolClient {
             task.state = .completed
             session.taskRegistry.remove(task)
         case .dataCompletionHandler(let completion):
-            let data = Data()
```
Comment (nethraravindran) [inline Foundation/NSURLSession/NSURLSessionTask.swift]: We will need to create this data if `protocol.properties[.responseData]` is nil.

**Comment by nethraravindran** [comment]
@pushkarnk  Thanks for this PR. Looks good to me :)

**Comment by pushkarnk** [comment]
Thanks @nethraravindran 

I've done that here:
```
completion(`protocol`.properties[URLProtocol._PropertyKey.responseData] as? Data ?? Data(), task.response, nil)
```

**Comment by pushkarnk** [comment]
A word about protocol-specific data in the URLProtocol objects. It seems that, ideally, `URLProtocol.property(forKey:in)` and `URLProtocol.setProperty(_:forKey:in)` should be used to store protocol specific data. However the comments methods indicate that this data must be stored in the URLRequest object (?). I am not sure if it makes sense to store response data/metadata in the request object. I am hence storing it in a dictionary in the protocol object. In future, once we have clarity on where to store it, we will need to make that change.

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Pull request merged**